livekit-agents/livekit/agents/avatar/avatar_worker.py

```diff
@@ -0,0 +1,173 @@
+import asyncio
+import logging
+from dataclasses import dataclass
+from typing import AsyncIterator, Coroutine, Optional, Protocol
+
+from livekit import rtc
+from livekit.agents import utils
+
+from ..pipeline.datastream_io import AudioFlushSentinel, DataStreamAudioReceiver
+
+logger = logging.getLogger(__name__)
+
+
+class VideoGenerator(Protocol):
+    async def push_audio(self, frame: rtc.AudioFrame | AudioFlushSentinel) -> None:
+        """Push an audio frame to the video generator"""
+
+    def clear_buffer(self) -> None | Coroutine[None, None, None]:
+        """Clear the audio buffer, stopping audio playback immediately"""
+
+    async def stream(
+        self,
+    ) -> AsyncIterator[
+        tuple[rtc.VideoFrame, Optional[rtc.AudioFrame]] | AudioFlushSentinel
+    ]:
+        """Continuously yield video frames, idle frames are yielded when no audio is available"""
+
+
+@dataclass
+class MediaOptions:
+    video_width: int
+    video_height: int
+    video_fps: float
+    audio_sample_rate: int
+    audio_channels: int
+
+
+# TODO(long): move to a plugin like livekit-plugins-avatar?
+class AvatarWorker:
```

Comment:
nit: perhaps don't call it a "worker" to avoid confusion with the "agent worker".

how about calling it `AvatarRunner` or `AvatarHandler`, etc